src/codeu/chat/database/Database.java

```diff
@@ -0,0 +1,55 @@
+package codeu.chat.database;
+
+import java.sql.*;
+
+import codeu.chat.util.Logger;
+
+public final class Database {
+
+  private final static Logger.Log LOG = Logger.newLog(Database.class);
+
+  private String m_dbPath;
+  private Connection m_connection;
+
+  public Database(String dbPath) {
+      m_dbPath = dbPath;
+  }
+
+  /**
+   * Attempts to validate the connection.
+   * If the connection is closed or does not exist, the function will try
+   * to reconnect.
+   *
+   * @return Whether the connection was succesfully validated.
+   */
+  private boolean validateConnection() {
```

Comment:
Might be good to call this something like `connectIfRequired()` instead, since this method does not seem to be validating much, rather creating a connection if one does not exist already or is not valid any more.